dashboard/src/containers/Dashboard.tsx

```diff
@@ -0,0 +1,276 @@
+import { Dispatch } from 'redux';
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { Bar } from 'react-chartjs-2';
+import {
+    Button, Card, CardHeader, CardBody,
+    Container, Row, Col } from 'reactstrap';
+
+import * as dispatchers from '../dispatchers';
+import { RootState } from '../types';
+import ParkingRegionsMap from './ParkingRegionsMap';
+import TimeSelect from './TimeSelect';
+import RegionSelector from  './RegionSelector';
+
+import ReactTable from 'react-table';
+import 'react-table/react-table.css';
+
+import './Dashboard.css';
+
+const bar = {
+    labels: ['16', '18', '20', '22', '0', '2', '4', '6', '8', '10', '12', '14'],
+    datasets: [
+        {
+            backgroundColor: 'rgba(255,99,132,0.2)',
+            borderColor: 'rgba(255,99,132,1)',
+            borderWidth: 1,
+            hoverBackgroundColor: 'rgba(255,99,132,0.4)',
+            hoverBorderColor: 'rgba(255,99,132,1)',
+            data: [108, 95, 75, 35, 25, 21, 28, 35, 89, 81, 92, 99]
+        }
+    ]
+};
+
+interface Props {
+    autoUpdate: boolean;
+    onUpdate: () => void;
+    onLogout: (event: React.MouseEvent<{}>) => void;
+}
+
+type TimerId = number;
+
+class Dashboard extends React.Component<Props> {
+    timer: TimerId|null = null;
+    timerInterval: number = 1000; // 1 second
+
+    componentDidMount() {
+        if (this.props.autoUpdate && !this.timer) {
+            this.enableAutoUpdate();
+        }
+    }
+
+    componentWillReceiveProps(nextProps: Props) {
+        if (nextProps.autoUpdate && !this.timer) {
+            this.enableAutoUpdate();
+        }
+        if (!nextProps.autoUpdate && this.timer) {
+            this.disableAutoUpdate();
+        }
+    }
+
+    enableAutoUpdate() {
+        this.autoUpdate();
+        if (this.timer) {
+            return;  // Was already enabled
+        }
+        this.timer = window.setInterval(
+            this.autoUpdate.bind(this), this.timerInterval);
+    }
+
+    disableAutoUpdate() {
+        if (!this.timer) {
+            return;  // Was already disabled
+        }
+        window.clearInterval(this.timer);
+        this.timer = null;
+    }
+
+    autoUpdate() {
+        if (this.props.onUpdate) {
+            this.props.onUpdate();
+        }
+    }
+
+    render() {// xl lg md sm xs
```

Comment:
I found this comment not helping me that much, I had no idea why the sizes were listed here since that is not the case in other render methods that has been written.

Reply:
Oops.. it was left there by accident.. were there just to remind me about the bootstrap classes which I'm not that familiar with yet... Will remove.